src/tree/path.js

```diff
@@ -1,51 +1,75 @@
+import isOnlyLeafs from '../utils/util';
+
 export default function path(node, o, isVertical) {
   // Create the lines (links) between the nodes
   node
     .append('path')
     .attr('class', 'link')
-    .attr('id', (d) => d.data.id)
-    .attr('d', (d) => {
+    .attr('id', d => d.data.id)
+    .attr('d', d => {
       if (d.parent) {
-        // starting at self, ending at parent
-        const start = { x: o.x(d) + o.pathOffsetSelf.x, y: o.y(d) + o.pathOffsetSelf.y };
-        const end = { x: o.x(d.parent) + o.pathOffsetParent.x, y: o.y(d.parent) + o.pathOffsetParent.y };
+        const halfNode = { x: o.nodeSize.width / 2, y: o.nodeSize.height / 2 };
+        const start = { x: o.x(d) + halfNode.x, y: o.y(d) + halfNode.y };
+        const end = { x: o.x(d.parent) + halfNode.x, y: o.y(d.parent) + halfNode.y };
         const size = { x: Math.abs(end.x - start.x), y: Math.abs(end.y - start.y) };
-        // factors inverseíng direction of lines and curves
         const inverse = { x: end.x - start.x < 0 ? -1 : 1, y: end.y - start.y < 0 ? -1 : 1 };
-        // radius of elbow
         const rDef = 30;
+
+        if (isOnlyLeafs(d.parent.children)) {
+          const halfDepth = o.depthSpacing / 2;
+          const r = { x: inverse.x * rDef, y: inverse.y * rDef };
+          if (isVertical) {
+            return `
+              M ${start.x} ${start.y}
+              L ${end.x - halfNode.x - r.x} ${start.y}
+              C ${end.x - halfNode.x} ${start.y} ${end.x - halfNode.x} ${start.y} ${end.x - halfNode.x} ${start.y + r.y}
+              L ${end.x - halfNode.x} ${end.y + halfDepth - r.y}
+              C ${end.x - halfNode.x} ${end.y + halfDepth} ${end.x - halfNode.x} ${end.y + halfDepth} ${end.x - halfNode.x - r.x} ${end.y + halfDepth}
+              L ${end.x + r.x} ${end.y + halfDepth}
+              C ${end.x} ${end.y + halfDepth} ${end.x} ${end.y + halfDepth} ${end.x} ${end.y + halfDepth + r.y}
+              L ${end.x} ${end.y}
+              `;
+          }
+          return `
+            M ${start.x} ${start.y}
+            L ${start.x} ${end.y - halfNode.y - r.y}
+            C ${start.x} ${end.y - halfNode.y} ${start.x} ${end.y - halfNode.y} ${start.x + r.x} ${end.y - halfNode.y}
+            L ${end.x + halfDepth - r.x} ${end.y - halfNode.y}
+            C ${end.x + halfDepth} ${end.y - halfNode.y} ${end.x + halfDepth} ${end.y - halfNode.y} ${end.x + halfDepth} ${end.y - halfNode.y - r.y}
+            L ${end.x + halfDepth} ${end.y + r.y}
+            C ${end.x + halfDepth} ${end.y} ${end.x + halfDepth} ${end.y} ${end.x + halfDepth + r.x} ${end.y}
+            L ${end.x} ${end.y}
+            `;
```

Comment:
Should be able to name these two cases and move them to their own function/file.

Reply:
I actually just had an idea of how to generalize the whole thing. I'll try that otherwise I'll brake out those cases somehow